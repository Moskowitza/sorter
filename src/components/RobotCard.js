import React from 'react';
import Card from 'react-bootstrap/Card';
import PropTypes from 'prop-types';

const RobotCard = ({ name, email, id }) => (
  <Card style={{ width: '18rem' }}>
    <Card.Img alt="robots" src={`https://robohash.org/${id}?size=200x200`} />
    <Card.Body>
      <Card.Title>{name}</Card.Title>
      <Card.Text>{email}</Card.Text>
    </Card.Body>
  </Card>
);

RobotCard.propTypes = {
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
export default RobotCard;
