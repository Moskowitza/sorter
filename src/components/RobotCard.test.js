import { shallow, mount, render } from 'enzyme';
import React from 'react';

import RobotCard from './RobotCard';

// const enzyme = require('enzyme');
// const Adapter = require('enzyme-adapter-react-16');

// enzyme.configure({ adapter: new Adapter() });
// eslint-disable-next-line no-undef
test('robotCard', () => {
  // eslint-disable-next-line no-undef
  const robotCard = shallow(<RobotCard
    id={1}
    name="aaron"
    email="aaron@email.com"
  />);
  expect(robotCard.length).toEqual(1);
});

test('robotCard Snapshot', () => {
  expect(shallow(<RobotCard
    id={1}
    name="aaron"
    email="aaron@email.com"
  />).debug()).toMatchSnapshot();
});
