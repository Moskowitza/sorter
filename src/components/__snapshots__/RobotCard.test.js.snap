// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`robotCard Snapshot 1`] = `
"<Card style={{...}} body={false}>
  <CardImg alt=\\"robots\\" src=\\"https://robohash.org/1?size=200x200\\" variant={{...}} />
  <CardBody>
    <CardTitle>
      aaron
    </CardTitle>
    <CardText>
      aaron@email.com
    </CardText>
  </CardBody>
</Card>"
`;
